src/cluster/brokerPool.js

```diff
@@ -163,6 +163,23 @@ module.exports = class BrokerPool {
 
         const replacedBrokers = []
 
+        // Update the seed broker if necessary
```

Comment:
Could you write a unit test that explains what this is doing and why? I understand this is a workaround for something in Oracle's streaming service, but I have no idea when it's necessary to update the seed broker or why.

Reply:
Thanks for checking this @Nevon. The bug fix here is specifically on line 195, which checks if the nodeId matches the seed broker's nodeId before returning it in the reduce call. Updating the seed broker is existing logic that I had to move around, but still works the same.

The current code does the seed broker nodeId and rack updates on lines 179-180: https://github.com/tulios/kafkajs/pull/1193/files#diff-017114e919136dd5728c2aa40fbab06e921c35272aa7d220f6f5e11afd29ee93L179-L180

Having checked the git history, it looks like this was done to resolve some connection issues, e.g. if the seed broker is recreated and has a new nodeId or rack. These commits were a few years ago, so it's hard to say if it's still needed or not. Do you think it can be removed?

Reply:
I tested removing this code and discovered several tests that failed. Turns out this code is required for the seed broker, since it starts off with just a host and port (no nodeId / rack): https://github.com/tulios/kafkajs/blob/master/src/cluster/brokerPool.js#L74-L77

I clarified this in the code a bit and added tests that target this.